tests/tasks/el_repo_setup.yml

```diff
@@ -1,4 +1,19 @@
 # SPDX-License-Identifier: BSD-3-Clause
+- name: Set the ansible_facts required by the network role test
+  set_fact:
+    network_test_required_facts:
+      - distribution
+      - distribution_major_version
+      - distribution_version
+      - os_family
+
+- name: Gather the minimum subset of ansible_facts required by the network
+    role test
+  setup:
+    gather_subset: min
+  when: not ansible_facts.keys() | list |
+    intersect(network_test_required_facts) == network_test_required_facts
```

Comment:
```yaml
  vars:
    network_test_required_facts:
      - distribution
      - distribution_major_version
      - distribution_version
      - os_family
```
unless you need `network_test_required_facts` to be a global variable, use it elsewhere